tests/release/release.go

```diff
@@ -0,0 +1,128 @@
+		BeforeAll(func() {
+			// Recreate random namespaces names per each test because if using same namespace names, the next test will not be able to create the namespaces as they are terminating
+			devNamespace = "user-" + uuid.New().String()
+			managedNamespace = "managed-" + uuid.New().String()
+
+			// Create the dev namespace
+			_, err := framework.CommonController.CreateTestNamespace(devNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating namespace '%s': %v", devNamespace, err)
+
+			// Create the managed namespace
+			_, err = framework.CommonController.CreateTestNamespace(managedNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating namespace '%s': %v", managedNamespace, err)
+		})
+
+		AfterAll(func() {
+			// Delete the dev and managed namespaces with all the resources created in them
+			Expect(framework.CommonController.DeleteNamespace(devNamespace)).NotTo(HaveOccurred())
+			Expect(framework.CommonController.DeleteNamespace(managedNamespace)).NotTo(HaveOccurred())
+		})
+
+		var _ = Describe("Creation of the 'Happy path' resources", func() {
+			It("Create an ApplicationSnapshot.", func() {
+				_, err := framework.ReleaseController.CreateApplicationSnapshot(snapshotName, devNamespace, applicationName, snapshotComponents)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a Release Strategy", func() {
+				_, err := framework.ReleaseController.CreateReleaseStrategy(releaseStrategyName, managedNamespace, releasePipelineName, releasePipelineBundle, releaseStrategyPolicy, ReleaseStrategySpecParams_extraConfigGitUrl, ReleaseStrategySpecParams_extraConfigPath, ReleaseStrategySpecParams_extraConfigRevision, releaseStrategyServiceAccount)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a ReleasePlan in dev namespace", func() {
+				_, err := framework.ReleaseController.CreateReleasePlan(sourceReleasePlanName, devNamespace, applicationName, managedNamespace, "")
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a ReleasePlanAdmission in managed namespace", func() {
+				_, err := framework.ReleaseController.CreateReleasePlanAdmission(targetReleasePlanAdmissionName, devNamespace, applicationName, managedNamespace, "", releaseStrategyName)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a Release", func() {
+				_, err := framework.ReleaseController.CreateRelease(releaseName, devNamespace, snapshotName, sourceReleasePlanName)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		var _ = Describe("Post-release verification", func() {
+			It("A PipelineRun should have been created in the managed namespace", func() {
+				Eventually(func() error {
+					_, err := framework.ReleaseController.GetPipelineRunInNamespace(managedNamespace, releaseName, devNamespace)
+
+					return err
+				}, 1*time.Minute, defaultInterval).Should(BeNil())
+			})
+
+			It("The PipelineRun has started, was done, and succeeded", func() {
```

Comment:
can we say "finished" instead of "was done"?

Reply:
Done